frontend/add-leads/add-edit-materials.component.tsx

```diff
@@ -0,0 +1,111 @@
+import { useCss } from "kremling";
+import React from "react";
+import css from "./add-edit-materials.css";
+import easyFetch from "../util/easy-fetch";
+import MaterialEntry from "./add-edit-materialEntry.component";
+
+export default function AddEditMaterials({
+  addMaterial,
+  deleteMaterial,
+  editMaterial,
+  materials,
+}) {
+  const [material, setMaterial] = React.useState({ name: "" });
+
+  function handleSubmit(evt) {
```

Comment:
This function should call `setAddingMaterial(true)` (a new state value) and then there should be a React effect that does the saving.